lib/core/task.toit

```diff
@@ -247,41 +245,37 @@ class Task_ implements Task:
   // Task state initialized by the VM.
   id_ := null
 
-  // Deadline and cancel support.
+  // Task state initialized by Toit code.
+  name_ := null
+  background_ := null
+
+  // Deadline and cancelation support.
   deadline_ := null
   is_canceled_ := null
   critical_count_ := null
 
   // If the task is blocked in a monitor, this reference that monitor.
   monitor_ := null
 
-  // All running tasks are chained together in linked list.
+  // All running tasks are chained together in doubly linked list.
   next_running_ := null
   previous_running_ := null
+
+  // Waiting tasks are chained together in a singly linked list.
   next_blocked_ := null
 
   // Timer used for all sleep operations on this task.
   timer_ := null
 
-  name := null
-
-  background := null
+  static STATE_RUNNING    /int ::= 0
+  static STATE_SUSPENDING /int ::= 1
+  static STATE_SUSPENDED  /int ::= 2
+  state_ := null  // TODO(kasper): Document this.
 
 // ----------------------------------------------------------------------------
 
 task_new_ lambda/Lambda -> Task_:
   #primitive.core.task_new
 
-task_transfer_ to/Task_ detach_stack:
+task_transfer_to_ to/Task_ detach_stack:
```

Comment:
It's otherwise not possible to know what the boolean for each `task_transfer_to_` call is.
```suggestion
task_transfer_to_ to/Task_ --detach_stack/bool:
```

That said: this relies on the fact that named parameters are added after positional parameters, which is an implementation detail.
Alternatives are to have a function with named parameter that redirects to one without, or to keep the code as is.

Reply:
I thought about this. Will keep as is for now.